plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java

```diff
@@ -286,4 +314,96 @@ public void dropSchema(ConnectorSession session, String schemaName)
     {
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support dropping schemas");
     }
+
+    private Set<AggregateFunctionRule> aggregateFunctionRules()
+    {
+        JdbcTypeHandle bigIntHandle = new JdbcTypeHandle(Types.BIGINT, Optional.of("bigint"), 0, 0, Optional.empty(), Optional.empty());
+        JdbcTypeHandle doubleHandle = new JdbcTypeHandle(Types.DOUBLE, Optional.of("double"), 0, 0, Optional.empty(), Optional.empty());
+        Set<Type> numericTypes = Set.of(DOUBLE, BIGINT, REAL);
+        Set<Type> numericAndTimeStampType = Set.of(DOUBLE, BIGINT, REAL, TimestampType.createTimestampType(3));
+        Set<Type> allTypes = Set.of(DOUBLE, BIGINT, REAL, TimestampType.createTimestampType(3), VARCHAR);
+        ImmutableSet.Builder<AggregateFunctionRule> builder = ImmutableSet.<AggregateFunctionRule>builder()
+                .add(new ImplementCountAll(bigIntHandle))
+                .add(new ImplementCount(bigIntHandle));
+
+        allTypes.forEach(type ->
+                builder.add(SingleInputAggregateFunction.builder()
+                        .prestoName("approx_distinct")
+                        .expression("approx_count_distinct(%s)")
+                        .jdbcTypeHandle(bigIntHandle)
+                        .outputType(BIGINT)
+                        .inputTypes(type)
```

Comment:
move input type after the name, since `(name, input type, output type?)` defines _what_ should be matched and the rest defines how it should be matched?

Reply:
Done